### PR TITLE
Fix config auto-generation

### LINKED ISSUE #130: Fix config auto-generation
I assume, some unproper merge lead to failing auto-generation of `SimonaConfig` class / wrong data types to be expected.

### LINKED ISSUE #148: Fix `genConfigSample` gradle task
Possible merge it with `genConfigClass`, since it requires parameter `--dd` for output anyway

### PR DESCRIPTION
**Pull request by ckittl**
Fixes #130 
Fixes #148

### CODE REVIEW

**Comment by johanneshiry** [comment]
1. CI is broken
2. I would prefer to know exactly where, that is in which PR this stuff has become broken. Doesn't feel well to approve & merge stuff where we _assume_ that this _might_ have been broken some time w/o knowing the source of this _apparently_ invalid code!

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
I just have some tscfg-related comments. The long comment could also be addressed in a separate PR.

Furthermore, `genConfigSample` currently does not work without specifying a destination directory for config classes (`--dd`). Looks like a bug in tscfg, I will investigate further.

File: gradle/scripts/tscfg.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ task genConfigClass {
       args = [
         "build/tscfg-${tscfgVersion}.jar",
         "--spec",
-        "src/main/resources/config/simona-config-template.conf",
+        "src/main/resources/config/config-template.conf",
```
[the remark this answers is not in the document]
Comment (ckittl) [reply gradle/scripts/tscfg.gradle:18]: However, I just recognize, that this feature is provided by #66. Don't know, why this part is yet apparent here... So, I'll remove `genExperimentsConfigClass` and then we should be good to go!

[the remark this answers is not in the document]
Comment (ckittl) [reply gradle/scripts/tscfg.gradle:18]: Apparently, I will add the documentation in that other PR.

**Review by sebastian-peter — DISMISSED** [review]

**Comment by sebastian-peter** [comment]
I still get a `FileNotFoundException: \tmp\ExampleCfg.java` when executing `genConfigSample`, is it just me?

File: gradle/scripts/tscfg.gradle [hunk 2]
```diff
@@ -29,35 +29,6 @@ task genConfigClass {
...
-        "--cn",
-        "ExperimentsConfig",
-        "--dd",
-        "src/main/scala/edu/ie3/simona/config/"
-      ]
-    }
-  }
-}
-
-
```
Comment (johanneshiry) [inline gradle/scripts/tscfg.gradle:60]: Why? Does not relate to #130 nor #148

File: src/main/scala/edu/ie3/simona/config/ConfigFailFast.scala [hunk 3]
```diff
@@ -396,10 +400,16 @@ case object ConfigFailFast extends LazyLogging {
       )
 
     voltLvls.foreach { voltLvl =>
-      {
-        if (!ConfigConventions.voltLvlRegex.matches(voltLvl))
+      Try(Quantities.getQuantity(voltLvl.vNom)) match {
+        case Success(quantity) =>
+          if (!quantity.getUnit.isCompatible(Units.VOLT))
+            throw new InvalidConfigParameterException(
+              s"The given nominal voltage '${voltLvl.vNom}' cannot be parsed to electrical potential!"
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/config/ConfigFailFast.scala:407]:
```suggestion
              s"The given nominal voltage '${voltLvl.vNom}' cannot be parsed to electrical potential! Please provide the volt level with its unit, e.g. \"20 kV\""
```

File: src/main/scala/edu/ie3/simona/config/ConfigFailFast.scala [hunk 4]
```diff
@@ -396,10 +400,16 @@ case object ConfigFailFast extends LazyLogging {
...
+            )
+        case Failure(exception) =>
           throw new InvalidConfigParameterException(
-            s"The definition string for voltLvl '$voltLvl' does not comply with the definition {<id>, <rated voltage>}!"
+            s"The given nominal voltage '${voltLvl.vNom}' cannot be parsed to Quantity.",
+            exception
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/simona/config/ConfigFailFast.scala:412]:
```suggestion
            s"The given nominal voltage '${voltLvl.vNom}' cannot be parsed to a quantity. Did you provide the volt level with it's unit (e.g. \"20 kV\")?",
            exception
```

**Review by johanneshiry — CHANGES_REQUESTED** [review]
Thanks for your work. Looks 90% good to me. You may consider my suggestions with a slightly increased verbosity (verbosity has been high already, but I thought an additional hint might be helpful). Additionally, I have one questions regarding the deleted part of the experiments config as I couldn't find anything about it in the linked issues. You might explain the removal. 

Thanks again :)

File: gradle/scripts/tscfg.gradle (hunk 2 above)
Comment (ckittl) [reply gradle/scripts/tscfg.gradle:60]: I just recognized in the [discussion with sebastian](https://github.com/ie3-institute/simona/pull/131#discussion_r809149780) that this is code, that was meant to be introduced with #66, but due to some error already is in code base right now.

**Comment by ckittl** [comment]
> I still get a `FileNotFoundException: \tmp\ExampleCfg.java` when executing `genConfigSample`, is it just me?
> 
> If not, #148 should probably be fixed in a different PR.

According to the error message, you quoted, I assume you are running the command in windows os?

**Comment by ckittl** [comment]
> > I still get a `FileNotFoundException: \tmp\ExampleCfg.java` when executing `genConfigSample`, is it just me?
> > If not, #148 should probably be fixed in a different PR.
> 
> According to the error message, you quoted, I assume you are running the command in windows os?

Okay... It was an issue with the default destination directory in tscfg. I handed in a PR at carueda/tscfg#151, that might make this workaround obsolete.

**Comment by ckittl** [comment]
> > > I still get a `FileNotFoundException: \tmp\ExampleCfg.java` when executing `genConfigSample`, is it just me?
> > > If not, #148 should probably be fixed in a different PR.
> > 
> > 
> > According to the error message, you quoted, I assume you are running the command in windows os?
> 
> Okay... It was an issue with the default destination directory in tscfg. I handed in a PR at [carueda/tscfg#151](https://github.com/carueda/tscfg/pull/151), that might make this workaround obsolete.

Merged, but not yet released.

**Review by sebastian-peter — APPROVED** [review]
With this workaround, `genConfigSample` works again, thank you. I created #178 to remove the workaround once it's not necessary anymore. So from my side, this is good to go.

**Comment by sebastian-peter** [comment]
This branch was just way out of sync with dev. A merge fixed everything.

**Review by t-ober — APPROVED** [review]

**Pull request merged**